### PR TITLE
docs: Created a Regression Getting Started Tutorial 

### LINKED ISSUE #400: Getting started guide for regression
Similar to the getting started guide for classification (based on the Titanic dataset, #330), we should have another getting started guide for regression (based on the House Sales dataset from [here](https://github.com/Safe-DS/Stdlib-Examples/blob/main/src/safeds_examples/tabular/_house_sales/data/house_sales.csv)). Copy the data to `docs/tutorials/data` and add another notebook using this data.

### PR DESCRIPTION
**Pull request by PhilipGutberlet**
Closes #400

### Summary of Changes

Created a Regression Getting Started Tutorial and added it together with the Classification Getting Started Tutorial to the docs.

### CODE REVIEW

File: docs/tutorials/getting_started_regression.ipynb [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 58,
+   "outputs": [],
+   "source": [
+    "feature_columns = set(train_table.column_names)-set([\"price\",\"id\"])\n",
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:77]:
```suggestion
    "feature_columns = set(train_table.column_names) - set([\"price\",\"id\"])\n",
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    "\n",
+    "tagged_train_table= train_table.tag_columns(\"price\", feature_names=[\n",
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:79]:
```suggestion
    "tagged_train_table = train_table.tag_columns(\"price\", feature_names=[\n",
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 56,
+   "outputs": [],
+   "source": [
+    "from safeds.data.tabular.containers import Table\n",
+    "\n",
+    "pricing = Table.from_csv_file(\"data/house_sales.csv\")\n",
+    "pricing.slice_rows(1,15)"
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:31]: Add comment saying that we slice the rows for overview purposes.

File: docs/tutorials/getting_started_regression.ipynb [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 60,
+   "outputs": [],
+   "source": [
+    "\n",
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:124]:
```suggestion
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 59,
+   "outputs": [],
+   "source": [
+    "from safeds.ml.classical.regression import DecisionTree\n",
+    "\n",
+    "model = DecisionTree()\n",
+    "fitted_model = model.fit(tagged_train_table)\n"
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:104]:
```suggestion
    "fitted_model = model.fit(tagged_train_table)"
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 58,
+   "outputs": [],
+   "source": [
+    "feature_columns = set(train_table.column_names)-set([\"price\",\"id\"])\n",
+    "\n",
+    "tagged_train_table= train_table.tag_columns(\"price\", feature_names=[\n",
+    "    *feature_columns\n",
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:80]:
```suggestion
    "    *feature_columns",
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+   "cell_type": "code",
+   "execution_count": 57,
+   "outputs": [],
+   "source": [
+    "split_tuple = pricing.split(0.60)\n",
+    "\n",
+    "train_table = split_tuple[0]\n",
+    "testing_table = split_tuple[1]\n",
+    "\n",
+    "test_table = testing_table.remove_columns([\"price\"])\n"
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:57]:
```suggestion
    "test_table = testing_table.remove_columns([\"price\"])"
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "tagged_test_table= testing_table.tag_columns(\"price\", feature_names=[\n",
+    "    *feature_columns\n",
+    "])\n",
+    "\n",
+    "fitted_model.mean_absolute_error(tagged_test_table)\n"
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:161]:
```suggestion
    "fitted_model.mean_absolute_error(tagged_test_table)"
```

**Review by alex-senger — CHANGES_REQUESTED** [review]

File: docs/tutorials/getting_started_regression.ipynb
Comment (lars-reimann) [inline docs/tutorials/getting_started_regression.ipynb:1]: For the calls of `slice_rows` I'd suggest to also include the first row (`slice_rows(0, 15)` or `slice_rows(end=15)`).

File: docs/tutorials/getting_started_classification.ipynb
[the remark this answers is not in the document]
Comment (lars-reimann) [reply docs/tutorials/getting_started_classification.ipynb:1]: Likewise, when creating the test set, it makes sense to shuffle the rows first.

File: docs/tutorials/getting_started_regression.ipynb
Comment (lars-reimann) [reply docs/tutorials/getting_started_regression.ipynb:1]: Likewise, when creating the test set, it makes sense to shuffle the rows first.

File: docs/tutorials/getting_started_regression.ipynb [hunk 9]
```diff
@@ -0,0 +1,189 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 56,
+   "outputs": [],
+   "source": [
+    "from safeds.data.tabular.containers import Table\n",
+    "\n",
+    "pricing = Table.from_csv_file(\"data/house_sales.csv\")\n",
+    "#For visualisation purposes we only print out the first 15 rows.\n",
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:31]:
```suggestion
    "# For visualisation purposes we only print out the first 15 rows.\n",
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 10]
```diff
@@ -0,0 +1,189 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 60,
+   "outputs": [],
+   "source": [
+    "prediction = fitted_model.predict(\n",
+    "    test_table\n",
+    ")\n",
+    "#For visualisation purposes we only print out the first 15 rows.\n",
```
Comment (alex-senger) [inline docs/tutorials/getting_started_regression.ipynb:127]:
```suggestion
    "# For visualisation purposes we only print out the first 15 rows.\n",
```

**Review by alex-senger — CHANGES_REQUESTED** [review]

**Review by alex-senger — DISMISSED** [review]
[LGTM](https://lgtm.party/p/978HG1SG/)

**Review by jxnior01 — DISMISSED** [review]
Nicee!!

File: docs/tutorials/getting_started_classification.ipynb [hunk 11]
```diff
@@ -0,0 +1,299 @@
...
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [],
+   "metadata": {
+    "collapsed": false
+   }
```
Comment (Marsmaennchen221) [inline docs/tutorials/getting_started_classification.ipynb:275]:
```suggestion
   }
```

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]

**Comment by zzril** [comment]
Not sure about the sorting of the tutorials (the `Data Processing` seems simpler to me and should possibly be first).

But can make a new issue for that.

**Review by zzril — CHANGES_REQUESTED** [review]
The numbering in the notebook seems off.

After `3. Tag the price Column as the target variable to be predicted. Use the new names of the fitted Columns as features, which will be used to make predictions based on the target variable.` follows `6. Use Decision Tree regressor as a model for the regression. Pass the "tagged_pricing" table to the fit function of the model:`.

**Comment by Marsmaennchen221** [comment]
> Not sure about the sorting of the tutorials (the `Data Processing` seems simpler to me and should possibly be first).
> 
> But can make a new issue for that.

I agree. I am also not sure about the "Getting started" in the title. All tutorials right now are some kind of a "Getting started" tutorial so I think we should just remove this prefix and rearrange the tutorials. We can do this in this pr

**Comment by Marsmaennchen221** [comment]
> Also, the code cell after `3.` looks strange:
> 
> ```python
> feature_columns = set(train_table.column_names) - set(["price", "id"])
> 
> tagged_train_table = train_table.tag_columns("price", feature_names=[
>     *feature_columns])
> ```
> 
> What's up with the `*`?
The `*` lists all the values in the given set so the list does not get one variable as set but gets all the values from the set as different variables:
```py
a = set(["a", "b", "c"])
print(a)
print([a])
print([*a])

"""
Output:
{'b', 'c', 'a'}
[{'b', 'c', 'a'}]
['b', 'c', 'a']
"""
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+    "    *feature_columns])\n"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "6. Use `Decision Tree` regressor as a model for the regression. Pass the \"tagged_pricing\" table to the fit function of the model:\n"
```
Comment (Marsmaennchen221) [inline docs/tutorials/getting_started_regression.ipynb:90]:
```suggestion
    "4. Use `Decision Tree` regressor as a model for the regression. Pass the \"tagged_pricing\" table to the fit function of the model:\n"
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+    "fitted_model = model.fit(tagged_train_table)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "7. Use the fitted decision tree regression model, that we trained on the training dataset to predict the price of a house in the test dataset.\n"
```
Comment (Marsmaennchen221) [inline docs/tutorials/getting_started_regression.ipynb:113]:
```suggestion
    "5. Use the fitted decision tree regression model, that we trained on the training dataset to predict the price of a house in the test dataset.\n"
```

File: docs/tutorials/getting_started_regression.ipynb [hunk 14]
```diff
@@ -0,0 +1,189 @@
...
+    "prediction.slice_rows(0,15)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "7. You can test the mean absolute error of that model with the initial testing_table as follows:\n"
```
Comment (Marsmaennchen221) [inline docs/tutorials/getting_started_regression.ipynb:137]:
```suggestion
    "6. You can test the mean absolute error of that model with the initial testing_table as follows:\n"
```

**Comment by zzril** [comment]
> The `*` lists all the values in the given set so the list does not get one variable as set but gets all the values from the set as different variables

Ok, thanks for the clarification.

**Comment by Marsmaennchen221** [comment]
> The numbering in the notebook seems off.
> 
> After `3. Tag the price Column as the target variable to be predicted. Use the new names of the fitted Columns as features, which will be used to make predictions based on the target variable.` follows `6. Use Decision Tree regressor as a model for the regression. Pass the "tagged_pricing" table to the fit function of the model:`.

Done

**Comment by zzril** [comment]
> > Not sure about the sorting of the tutorials (the `Data Processing` seems simpler to me and should possibly be first).
> > But can make a new issue for that.
> 
> I agree. I am also not sure about the "Getting started" in the title. All tutorials right now are some kind of a "Getting started" tutorial so I think we should just remove this prefix and rearrange the tutorials. We can do this in this pr

Makes sense.

Need to remember to update the reference in `mkdocs.yml` then,

**Review by jxnior01 — APPROVED** [review]

**Comment by zzril** [comment]
Nice, thanks.

Let me check the page in the browser really quick...

**Comment by zzril** [comment]
`Classification` also has a counting error (7 follows 7). :D

File: docs/tutorials/classification.ipynb [hunk 15]
```diff
@@ -181,7 +185,7 @@
   {
    "cell_type": "markdown",
    "source": [
-    "7. You can test the accuracy of that model as follows:"
+    "7. You can test the accuracy of that model with the initial testing_table as follows:"
```
Comment (Marsmaennchen221) [inline docs/tutorials/classification.ipynb:188]:
```suggestion
    "8. You can test the accuracy of that model with the initial testing_table as follows:"
```

**Review by zzril — APPROVED** [review]
[LGTM](https://lgtm.party/p/16637JCD/)
